packages/redux/src/epics/content.ts

```diff
@@ -16,11 +27,66 @@ export const getItemEpic =
     switchMap(({ payload }) =>
       ContentStoreService.getItem(payload, crafterConf.getConfig())
         .pipe(
-          tap(item => console.log(item)), // TODO remove tap once no longer needed
+          tap(item => console.log("GET_ITEM_EPIC", item)),
           map(item => getItemComplete(item))
         ))
   );
 
-export const allContentEpics = [
-  getItemEpic
-];
+export const getDescriptorEpic =
+  (action$: Observable<AnyAction>) => action$.pipe(
+      ofType(GET_DESCRIPTOR),
+      switchMap(({ payload }) =>
+      ContentStoreService.getDescriptor(payload, crafterConf.getConfig())
+          .pipe(
+              map(descriptor => getDescriptorComplete(descriptor))
+          ))
+  );
+
+export const getChildrenEpic =
+  (action$: Observable<AnyAction>) => action$.pipe(
+      ofType(GET_CHILDREN),
+      switchMap(({ payload }) =>
+      ContentStoreService.getChildren(payload, crafterConf.getConfig())
+          .pipe(
+              map(children => getChildrenComplete(children))
+          ))
+  );
+
+export const getTreeEpic =
+  (action$: Observable<AnyAction>) => action$.pipe(
+      ofType(GET_TREE),
+      switchMap(({ payload }) =>
+      ContentStoreService.getTree(payload.url, payload.depth, crafterConf.getConfig())
```

Comment:
Check on undefined & null behaviours for optional parameters (all service calls). What does the back end support/expect too.